src/main/java/core/basesyntax/strategy/ReturnHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class ReturnHandler implements OperationHandler {
+    private static final String RETURN = "r";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(RETURN)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
```

Comment:
The variable `oldQuantity` is being reset inside the loop, which seems incorrect as it would only store the quantity from the last iteration. The logic for updating the quantity of fruits seems flawed, as it should accumulate the quantity of the 'return' transactions rather than overwrite it.